nrcan_etl/src/energuide/embedded/region.py

```diff
@@ -20,22 +57,68 @@ class Region(enum.Enum):
 
     @classmethod
     def _from_name(cls, name: str) -> typing.Optional['Region']:
-        snake_name = name.upper().replace(' ', '_')
-        return Region[snake_name] if snake_name in Region.__members__ else None
+        english_region = Region[name] if name in Region.__members__ else None
+        if english_region:
+            return english_region
+
+        if name in _FRENCH_NAMES:
+            return Region[_FRENCH_NAMES[name]]
+
+        if name in _ALTERNATIVE_NAMES:
+            return Region[_ALTERNATIVE_NAMES[name]]
+
+        return None
+
 
     @classmethod
     def _from_code(cls, code: str) -> typing.Optional['Region']:
-        code = code.upper()
+        code = code.replace('.', '')
         for region in Region:
-            if code == region.value:
+            if code == region.value or _ALTERNATIVE_CODES.get(code) == region.value:
                 return region
+
+        return None
+
+
+    @classmethod
+    def _from_fuzzy_match(cls, data: str) -> typing.Optional['Region']:
+        # Name is prefix of full name
+        for region in Region.__members__:
+            if region.startswith(data):
+                LOGGER.warning(f'PREFIX: Interpreting {data} as {region}')
+                return Region[region]
+
+        # Code separated by _
+        code = data.replace('_', '')
+        region = cls._from_code(code)
+        if region:
+            return region
+
+        # Levenshtein minimum number of edits to closest region
+        ratios = [(Levenshtein.ratio(region, data), region) for region in Region.__members__.keys() - {'UNKNOWN'}]
+        ratio, region = max(ratios)
+
+        if ratio > _MINIMUM_SIMILARITY:
+            LOGGER.warning(f'SIMILARITY: Interpreting {data} as {region}')
```

Comment:
should this be a warning or info? isn't this the expected behaviour if the data is clean?

Reply:
If the data is clean we should never get to this point, and it should exactly match either a province/territory name or its abbreviation.

I think this should be a warning since it's possible that this picks the wrong province, and it should be logged that we had to approximate it.